core/module_specifier.rs

```diff
@@ -46,6 +46,10 @@ impl fmt::Display for ModuleResolutionError {
 pub struct ModuleSpecifier(Url);
 
 impl ModuleSpecifier {
+  fn is_dummy_specifier(specifier: &str) -> bool {
+    specifier == "<unknown>"
```

Comment:
I think this is good as a short term fix.

Longer term, we should make ModuleSpecifier an enum , something like `enum ModuleSpecifier { Url(Url), Repl(PathBuf) }`, and stop converting between String/Url/ModuleSpecifier all the time, and change the resolve_import signature to 
`pub fn resolve_import(
    specifier: &str,
    base: ModuleSpecifier
  ) -> Result<ModuleSpecifier, ModuleResolutionError>`.